src/compiler/checker.ts

```diff
@@ -7301,11 +7318,8 @@ namespace ts {
                 if (type.flags & TypeFlags.Union) {
                     return getUnionType(map((<UnionType>type).types, getWidenedConstituentType));
                 }
-                if (isArrayType(type)) {
-                    return createArrayType(getWidenedType((<TypeReference>type).typeArguments[0]));
-                }
-                if (isTupleType(type)) {
-                    return createTupleType(map(type.elementTypes, getWidenedType));
+                if (isArrayType(type) || isTupleType(type)) {
+                    return createTypeReference((<TypeReference>type).target, map((<TypeReference>type).typeArguments, getWidenedType));

```

Comment:
><TypeReference> [](start = 48, length = 15)

Could you make `isArrayType` and `isTupleType` both return `x is TypeReference`? It would clean up this line, although I don't know if the overall effect would be less confusing.